tsl/src/nodes/columnar_scan/columnar_scan.c

```diff
@@ -0,0 +1,805 @@
+/*
```

Comment:
Last time we discussed this, there was an idea to put all the hyperstore files into a separate directory, let's put this one there as well? For ease of tweaking workflows and review requirements.

Reply:
Just following existing pattern of putting scan nodes under `nodes/`.  So, yes, hyperstore is split across two directories instead of one in order to not break the existing pattern followed by every other feature.

Are you thinking of a top-level directory that recreates the same pattern below the top-level? Note that we currently have a top-level directory `tsl/` for licensing purposes. If we'd create a top-level `hyperstore` directory it would make licensing less clear.

I don't think having two directories for Hyperstore is a big deal or problem.  But it is easy to change and I am not sure it even has to change for this merge and before we agree.

Reply:
Let's put the columnar scan into `tsl/hyperstore/nodes`? The difference with other features is that Hyperstore is a separate project with different development process and ownership, that just happens to live in the TSL extension for convenience.

Reply:
That makes no practical difference at all. The nodes are already in separate directories and making drastic restructuring of the repository for cosmetic reasons does not really make sense and only adds problems in maintaining the code, especially with regard to licensing issues that @erimatnor mentions.

Reply:
> Let's put the columnar scan into `tsl/hyperstore/nodes`? The difference with other features is that Hyperstore is a separate project with different development process and ownership, that just happens to live in the TSL extension for convenience.

After merge it will have the same development process and ownership as the rest of the code in the repo.

Reply:
>  After merge it will have the same development process and ownership as the rest of the code in the repo.

I heard a different thing last time we discussed it -- we're putting the Hyperstore into the TSL extension for convenience, but it remains a separate project with its own review requirements, team and so on. I'm not sure the Core DB team has the resources to take ownership of this pretty complicated experimental project, that was developed separately. Probably we have to return to this discussion after we figure out the 2.16 situation.